### PR TITLE
instr(txnames): Log statsd metrics for transaction name changes

### PR DESCRIPTION
**Pull request by jjbayer**
Log changes made to the transaction name by the `TransactionProcessor` to get an idea of what percentage of transactions is affected, and by which mechanism.

We can consider this a temporary metric and delete it once we are done with this initiative.

#skip-changelog

### CODE REVIEW

File: relay-server/src/statsd.rs [hunk 1]
```diff
@@ -479,6 +479,16 @@ pub enum RelayCounters {
...
+    /// The number of transaction events processed grouped by transaction name modifications.
+    /// This metric is tagged with:
+    ///  - `source_in`: The source of the transaction name before normalization.
+    ///    See the [transaction source
+    ///    documentation](https://develop.sentry.dev/sdk/event-payloads/properties/transaction_info/)
+    ///    for all valid values.
+    ///  - `change`: The mechanism that changed the transaction name.
+    ///    Either `"none"`, `"pattern"`, `"rule"`, or `"both"`.
+    ///  - `source_out`: The source of the transaction name after normalization.
+    TransactionNameChanges,
```
Comment (jjbayer) [inline relay-server/src/statsd.rs:470]: I considered adding a `set` metric as well to get the % of _unique_ names scrubbed, but I did not want to risk overwhelming statsd with that.

File: relay-server/src/utils/statsd.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+use relay_general::protocol::Event;
+use relay_general::types::{Annotated, RemarkType};
+
+use crate::statsd::RelayCounters;
+
+/// Log statsd metrics about transaction name modifications.
+///
+/// We have to look at event & meta before and after the modification is made,
+/// so we delegate to `f` in the middle of the function.
+pub fn log_transaction_name_metrics<F, R>(event: &mut Annotated<Event>, mut f: F) -> R
```
Comment (jjbayer) [inline relay-server/src/utils/statsd.rs:11]: This function turned out rather convoluted. We cannot log metrics from `relay-general`, so we have to analyze the event and its meta from `relay-server` to determine what changes were made.

File: relay-server/src/utils/statsd.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+where
+    F: FnMut(&mut Annotated<Event>) -> R,
+{
+    let Some(inner) = event.value() else {
+            return f(event);
+        };
```
Comment (olksdr) [inline relay-server/src/utils/statsd.rs]: Formatting a little bit off.

**Review by olksdr — APPROVED** [review]

File: relay-server/src/utils/statsd.rs (hunk 3 above)
Comment (jjbayer) [reply relay-server/src/utils/statsd.rs]: Thanks, interesting that the auto formatter did not catch it.

Comment (jjbayer) [inline relay-server/src/utils/statsd.rs]:
```suggestion
    let Some(inner) = event.value() else {
        return f(event);
    };
```

**Pull request merged**